Create tag version with 3 roles: load, identify and load + identify (#18)
fingerprintjs/fingerprintjs-pro-gtm: README.md

  </a>
</p>

# Fingerprint Pro Google Tag Manager template

> You can find an example of using this integration on [our website](https://dev.fingerprint.com/docs/fingerprintjs-pro-google-tag-manager)

Reviewer: ```suggestion
This repository contains a Google Tag Manager template you can use to add [Fingerprint Pro](https://fingerprint.com/) to your website.  

For step-by-step instructions on using this integration, see the full [Google Tag Manager guide](https://dev.fingerprint.com/docs/fingerprintjs-pro-google-tag-manager) in the Fingeprint Pro documentation.
```

## Usage

1. You should have Fingerprint Pro account
2. Add Fingerprint Pro Tag in your GTM admin panel (you can import it from [github repo](https://github.com/fingerprintjs/fingerprintjs-pro-gtm/blob/master/template.tpl))
3. Set up the public API key and choose the region
4. Use `FingerprintJSPro.loaded` event to get data from dataLayer. You can create trigger for this event.

> 5. Use `Result custom name` field to change the variable name for the result.

Reviewer: ```suggestion
1. [Sign up](https://dashboard.fingerprint.com/signup) for a Fingerprint Pro account if you haven't already.
2. Import this [template](https://github.com/fingerprintjs/fingerprintjs-pro-gtm/blob/master/template.tpl) into your Google Tag Manager workspace.
3. Add a Fingerprint Pro tag to your website. You will need your public API key and application region.
4. Use the `FingerprintJSProResult` or your own *Custom result name* to access the Fingerprint Pro result in GTM's `dataLayer`.
5. Use `FingerprintJSPro.loaded` and `FingerprintJSPro.identified` to create custom events and trigger actions after the JS agent is loaded or the visitor is identified. 
```

## Template Fields

> You can find more info on the Fingerprint Pro agent documentation page [https://dev.fingerprint.com/docs/js-agent](https://dev.fingerprint.com/docs/js-agent)

Reviewer: ```suggestion
For more information and the full API reference, see [Fingerprint Pro JS Agent](https://dev.fingerprint.com/docs/js-agent) in our documentation.
```

`Tag type` – The way you want to use the tag. There are 3 options:
  - `Load and identify` – the default behavior. Load the JS agent and identify the browser immediately. If you want to load the JS agent first and identify the browser later based on some event, use two separate `Load` and `Identify` Fingerprint Pro tags. 
  - `Load` – only load the JS Agent.
  - `Identify` – identify the browser. The JS Agent must be loaded before. You can collect additional metadata data first and then trigger this tag with the metadata inside `linkedId` or `tag`.

`Public API key` - Your public API key that authenticates the agent with the API.

> 

Reviewer: Line 41: "with Custom subdomain setup or Proxy integration."

  </a>
</p>

# Fingerprint Pro Google Tag Manager template

> This repository contains a Google Tag Manager template you can use to add [Fingerprint Pro](https://fingerprint.com/) to your website.  

Reviewer 2: Was it added to my fingerprint

For step-by-step instructions on using this integration, see the full [Google Tag Manager guide](https://dev.fingerprint.com/docs/fingerprintjs-pro-google-tag-manager) in the Fingeprint Pro documentation.

## Usage

1. [Sign up](https://dashboard.fingerprint.com/signup) for a Fingerprint Pro account if you haven't already.
2. Import this [template](https://github.com/fingerprintjs/fingerprintjs-pro-gtm/blob/main/template.tpl) into your Google Tag Manager workspace.
3. Add a Fingerprint Pro tag to your website. You will need your public API key and application region.
4. Use the `FingerprintJSProResult` or your own *Custom result name* to access the Fingerprint Pro result in GTM's `dataLayer`.
5. Use `FingerprintJSPro.loaded` and `FingerprintJSPro.identified` to create custom events and trigger actions after the JS agent is loaded or the visitor is identified. 

> ## Template Fields

Reviewer 2: 2c97d3e4caf95423a961554caa25ed81a1ad9c39

1. FingerprintJS Pro Google Tag Manager template

> 2. FingerprintJS Pro adapter for Google Tag Manager

Reviewer: Who would this be useful for? It seems like an implementation detail from my POV, considering that this integration is mostly aimed at less technical users. Have you considered putting it inside a separate README, or making it less prominent?

You can find more info on the FingerprintJS Pro agent documentation page [https://dev.fingerprint.com/docs/js-agent](https://dev.fingerprint.com/docs/js-agent)

> `Tag type` - The way you want to use the tag. It is 3 options:

Reviewer 3: ```suggestion
`Tag type` - The way you want to use the tag. There are 3 options:
```

>   - `Load and identify` – default behavior. Load JS Agent and instantly identify browser.

Reviewer 3: ```suggestion
  - `Load and identify` – default behavior. Load the JS Agent and identify the browser immediately.
```

>   - `Load` – just load JS Agent.

Reviewer 3: ```suggestion
  - `Load` – load the JS Agent only.
```

>   - `Identify` - identify browser. JS Agent needs to be loaded before. You can collect necessary data, then trigger this tag and provide `linkedId` or `tag`.

Reviewer 3: ```suggestion
  - `Identify` - identify the browser. The JS Agent must be loaded before. Collect necessary data, first, and then trigger this tag and provide `linkedId` or `tag`.
```

>   - `Identify` - identify browser. JS Agent needs to be loaded before. You can collect necessary data, then trigger this tag and provide `linkedId` or `tag`.

Reviewer: "JS Agent needs to be loaded before" I don't understand. If I want to load the agent and then identify later based on some event, do I need to have two separate tags, one with `load` and the other with `identify`? Or can I `load` and `identify` from a script programmatically?

Author: Yes `Identify` tag requires a `Load` tag. Load JS agent programmatically will not work because GTM tag requires special wrapper.

But if the customer loaded Agent using `Load` or `Load and identify` tag, it is possible to use the agent programmatically without adding a script tag.

Just as in documentation snippet:

```js
var fpPromise = FingerprintJS.load()

  // Analyze the visitor when necessary.
  fpPromise
    .then(function (fp) { return fp.get() })
    .then(result => console.log(result.requestId, result.visitorId))
```

> 

Reviewer: Github doesn't allow you to comment on any line in the file just the changed ones? Weird. I wanted to comment on line 96, it seems `scriptUrlPattern` is supported now, should we remove it from Limitations?

> `Tag type` - The way you want to use the tag. There are 3 options:
>   - `Load and identify` – default behavior. Load the JS Agent and identify the browser immediately.
>   - `Load` – load the JS Agent only.
>   - `Identify` - identify the browser. The JS Agent must be loaded before. Collect necessary data, first, and then trigger this tag and provide `linkedId` or `tag`.

Reviewer: ```suggestion
`Tag type` – The way you want to use the tag. There are 3 options:
  - `Load and identify` – the default behavior. Load the JS agent and identify the browser immediately. If you want to load the JS agent first and identify the browser later based on some event, use two separate `Load` and `Identify` Fingerprint Pro tags. 
  - `Load` – only load the JS Agent.
  - `Identify` – identify the browser. The JS Agent must be loaded before. You can collect additional metadata data first and then trigger this tag with the metadata inside `linkedId` or `tag`.
```

> `Result custom name` - you can specify the result field name in `dataLayer`

Reviewer: ```suggestion
`Result custom name` - you can specify the result field name in `dataLayer`.
```
There is a missing dot in the linkedId sentence too
